infrastructure/mail/mail.go

```diff
@@ -14,17 +15,31 @@ type (
 	// Mailer encapsulates the dependency
 	Mailer struct {
 		*mail.SMTPClient
-		Enabled    bool
+		Defaults   Defaults
 		DomainName string
 	}
 
+	MailerInit struct {
```

Comment:
NITPICK: could you add comments explaining what the difference between `Mailer` and `MailerInit` is

Reply:
Fixed